code/modules/heavy_vehicle/equipment/medical.dm

```diff
@@ -263,6 +264,76 @@
 	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
 	restricted_software = list(MECH_SOFTWARE_MEDICAL)
 
+/obj/item/device/healthanalyzer/mech //Used to set up the full body scan feature
+	var/obj/machinery/body_scanconsole/internal_bodyscanner = null
+	var/fullScan = FALSE //Toggle whether to do full or basic scan
+	
+/obj/item/device/healthanalyzer/mech/Initialize()
+	. = ..()
+	if(!internal_bodyscanner)
+		var/obj/machinery/body_scanconsole/S = new (src)
```

Comment:
Instead of doing this and assigning it further down to internal bodyscanner, do this:
`internal_bodyscanner = new /obj/machinery/body_scanconsole(src)`